src/JSONFileReader.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "Board.hpp"
+
+using json = nlohmann::json;
+
+class JSONFileReader
+{
+    json j_;
+public:
+    explicit JSONFileReader(std::string path);
+    int readRowsNumber();
+    int readColsNumber();
+    Lines readRows();
+    Lines readCols();
+};
+
+class InvalidDimensions : public std::exception
```

Comment:
Whole exception class can be implemented in this way

```suggestion
struct InvalidDimensions : public std::out_of_range
{
    InvalidDimensions() : std::out_of_range("Number of lines is different than number of cols/rows or number of cols/rows must be at least 3") {}
};
```